sphinx_gallery/gen_rst.py

```diff
@@ -498,7 +499,8 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
+    fname_pattern = gallery_conf.get('fname_pattern', 'plot_*')
+    if os.path.join(src_dir, fname) not in glob.glob(fname_pattern):
```

Comment:
Although probably more exact this seems like an overcomplicated translation to me. Any of `plot_.*`, `plot_.+`, `plot_[\w.]+` seems good enough to me.

For the record, I am not even sure why the underscore is escaped and why they use the flags MULTILINE and DOTALL.